ciao-scheduler/scheduler.go

```diff
@@ -183,6 +204,7 @@ func connectController(sched *ssntpSchedulerServer, uuid string) {
 
 	if sched.controllerMap[uuid] != nil {
 		glog.Warningf("Unexpected reconnect from controller %s\n", uuid)
+		sched.controllerMutex.Unlock()
```

Comment:
Yep.  I refactored that at some point to use a defer and didn't pull all the other unlocks.  Fixing.